### ISSUE TITLE
Generated data from GANGenerator() gives worse result in training for regression task

### ISSUE THREAD

**Issue by jomariya23156** [Diyago/Tabular-data-generation#21]
I used the data generated from GANGenerator in training and it gives worse results for the regression task I'm doing. I tried both default parameters and some custom parameters replicating what I saw in the example of this repo. Both of them don't work well. Here is the code I created the model:
`new_train, new_target = GANGenerator(gen_x_times=0.5, cat_cols=None, bot_filter_quantile=0.001,
                                       top_filter_quantile=0.999,
                                       is_post_process=True,
                                       adversaial_model_params={
                                           "metrics": "MSE", "max_depth": 15,
                                           "max_bin": 100, "n_estimators": 600,
                                           "learning_rate": 0.001, "random_state": 42,
                                       }, pregeneration_frac=2,
                                       epochs=300).generate_data_pipe(train, target,
                                                                      test, deep_copy=True,
                                                                      only_adversarial=False,
                                                                      use_adversarial=True,
                                                                      only_generated_data=True)`
One thing I noticed is that my training is stopped at around 90-120 epochs, even I have set it to be 300. I'm not sure whether I set some parameters poorly. So, I want some suggestions on how should I define parameters for this model to work with regression tasks. By the way, my data has already been preprocessed and cleaned containing one-hot encoded features and scaled features.

**Comment by Diyago** [comment]
Could you please send data stats you pass on to model this way:

df.describe()

**Comment by jomariya23156** [comment]
Here you are. The "metal_linker", "organic_linker1" and "organic_linker2" are categorical. The last column "CO2_working_capacity" is the target. These are the stats after dropping some NaNs and invalid values and before doing one-hot encoding and feature scaling. I have tried both versions with feature scaling and without feature scaling. Both produced bad newly generated data and decrease the score.

![TMLCC_detail_3_after](https://user-images.githubusercontent.com/73450708/136703924-ba7d4603-f43c-4df4-986a-7a5fea85b011.jpg)

![TMLCC_detail_4_after](https://user-images.githubusercontent.com/73450708/136703927-84779dc9-c3e9-4899-b096-2d3e0fe90a7b.jpg)

**Comment by Diyago** [comment]
1. Dummy training I just tried - works, but I suggest you change lightgbm params training:
2. "Accuracy may be bad since you didn't set num_leaves and 2^max_depth > num_leaves" - try setting max_depth = 2 or 3
3. What makes you feel that regression task quality much lower?

**Comment by Diyago** [comment]
4. Added params for early stopping - ```gan_params = {"batch_size": 500, "patience": 25, "epochs" : 500,}```

Try it out by installing new library version:
```!pip install tabgan==1.1.1```

**Comment by Diyago** [comment]
Reopen if problems still persists  and provide details for point "GANGenerator in training and it gives worse results for the regression"

Sry for late answer
